### PR TITLE
Add pygame.joystick support.

### PR DESCRIPTION
**Pull request by hodgestar**
This is a start on adding joystick support (see #8).

It requires the CFFI out-of-line build support pull request to land first (see #10).

### CODE REVIEW

**Comment by hodgestar** [comment]
As far as I can take this without a joystick. Ready for review and trying out.

File: pygame/joystick.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        """ get_id() -> int
+        get the Joystick ID
+        """
+        return self._id
+
+    def get_name(self):
+        """ get_name() -> string
+        get the Joystick system name
+        """
+        return sdl.SDL_JoystickName(self._id)
```
Comment (drnlm) [inline pygame/joystick.py]: This should be wrapped in a ffi.string(), otherwise the result really isn't that useful

**Comment by drnlm** [comment]
One comment, otherwise works fine for me with a couple of different pygame test programs.

**Comment by hodgestar** [comment]
Ready for re-review.

Pygame has some joystick tests but they are all marked `todo`. Worthwhile bring them across or not really?

**Comment by drnlm** [comment]
Looks good

:+1:

**Pull request merged**